pyasn1/codec/ber/decoder.py

```diff
@@ -115,7 +125,7 @@ def _asSeekableStream(substrate):
         If the supplied substrate cannot be converted to a seekable stream.
     """
     if isinstance(substrate, bytes):
-        return BytesIO(substrate)
+        return _CachingStreamWrapper(BytesIO(substrate))
```

Comment:
Should we always wrap input stream? Perhaps, no? The only reason for that here is to pass all reading through `_CachingStreamWrapper.read` generator routine.